### DISCUSSION TITLE
Lightweight charts not plotting y-axis with own dataset

### DISCUSSION THREAD

**Discussion by TheOxfordDeveloper** [tradingview/lightweight-charts#1560] [Q&A] [answered]
Hi, 

Here is the code to plot a static candlestick lightweight chart using your own csv file: 

html file:
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div id="tvchart"></div>
</body>
<script src="https://unpkg.com/lightweight-charts/dist/lightweight-charts.standalone.production.js"></script>
<script src="OHLCV_CSV_STATIC.js"></script>
</html>

js file: 
const getData = async () => {
  const res = await fetch('ohlcv_new.csv');
  const resp = await res.text();
  //   console.log(resp);
  const cdata = resp.split('\n').map((row) => {
    const [time1, open, high, low, close] = row.split(',');
    return {
      time: new Date(`${time1}`).getTime() / 1000,
      open: open * 1,
      high: high * 1,
      low: low * 1,
      close: close * 1,
    };
  });
  return cdata;
  //   console.log(cdata);
};

// getData();

const displayChart = async () => {
  const chartProperties = {
    width: 1500,
    height: 600,
    timeScale: {
      timeVisible: true,
      secondsVisible: false,
    },
    priceScale: {
      position: 'right', // Position of the price scale
      mode: 0, // Normal mode
      scaleMargins: {
        top: 25,
        bottom: 0.5,
      },
    },  
  };

  const domElement = document.getElementById('tvchart');
  const chart = LightweightCharts.createChart(domElement, chartProperties);
  const candleseries = chart.addCandlestickSeries();
  const klinedata = await getData();
  candleseries.setData(klinedata);
};

displayChart();

When I run the code it shows a beautiful candlestick chart in my browser (see image attached), however, the only thing that's missing is the y-axis. Does anyone know how I can get the y-axis to plot with the values on it? I have attached another image of a candlestick chart which DOES show the y-axis. This is what I want on my chart. 

The code that produces the candlestick chart with the y-axis shown on the chart doesn't import the data from a csv file, it just lists the data as a long list of data points inside brackets (), whereas in the code above I have imported the data as a csv file. 

I am writing my code on Visual Studio Code on Mac. 

Thanks 

![niwfndks 001](https://github.com/tradingview/lightweight-charts/assets/166934244/36e04ca2-93f4-4564-82cb-d89d4e51f3a6)

![yaxis 001](https://github.com/tradingview/lightweight-charts/assets/166934244/88ea3113-2dbf-463a-a7bf-574ded4e4b52)

**Comment by AdvMaple** [comment]
I can see the y-axis on the right hand side of the chart, in your image.
Could you draw on the image to clarify what you want display?

**Reply by TheOxfordDeveloper** [reply]
Thanks for your reply! I've attached two screenshots of two different charts. One has a y-axis on the right hand side of the chart and one doesn't. The one that doesn't is the chart I created using the code attached with my own financial dataset. The one that does have a y axis is the chart I created using the code attached but with a trading view financial dataset. I want to have a y-axis on the chart that I created using my own dataset. I have drawn on the image to clarify, and have also attached the image of the trading view chart that does have a y-axis to illustrate what I am trying to achieve with my own chart. 

Thank you 
![322215565-36e04ca2-93f4-4564-82cb-d89d4e51f3a6](https://github.com/tradingview/lightweight-charts/assets/166934244/47c721a2-4fdd-4ef3-aecb-e39978030b1b)
![322215615-88ea3113-2dbf-463a-a7bf-574ded4e4b52](https://github.com/tradingview/lightweight-charts/assets/166934244/adf5b9fc-49d8-474d-bef8-fdadf43ca9fd)

**Reply by AdvMaple** [reply]
can you send the sample csv file? @EMChloe

**Reply by AdvMaple** [reply]
And if you could, can you make a reproducing codesandbox?
I try to make one based on the the code you provide but the y-axis still showing.

![image](https://github.com/tradingview/lightweight-charts/assets/46006210/f09101d4-cbe7-4c6c-bf89-0233ead4dcf7)

This might cause by css, you might want to check your css, the page might be overflow and hidden away the y-axis

**Reply by TheOxfordDeveloper** [reply]
Hi, here is the dataset! 

I'm still new to all of this but I think I know how to check my CSS. I don't know how to make a reproducing codesandbox but I can try figure it out shortly (just about to head out right now). Thanks so much for your support so far (: 

The columns from left to right: Date ("time1" in code), open, high, low, close. I deleted the column titles because I thought this was making the code not run properly 


[ohlcv_new.csv](https://github.com/tradingview/lightweight-charts/files/14977809/ohlcv_new.csv)

**Reply by TheOxfordDeveloper** [reply]
Also, very interesting and reassuring that the code executes perfectly fine and as expected for you!

**Reply by AdvMaple** [reply, marked as answer]
![image](https://github.com/tradingview/lightweight-charts/assets/46006210/d2b800e2-f8d2-4074-b498-96e8f3a56c0a)

I imported the data and the y-axis is there.
Here the codesandbox: [link](https://codesandbox.io/p/sandbox/unruffled-keldysh-lzqvpp?file=%2Fsrc%2Findex.mjs%3A3%2C52)

**Reply by TheOxfordDeveloper** [reply]
Hi @AdvMaple thank you so much for your patience and for helping me. I realised that the error was that I simply did not scroll to the right far enough to see the y-axis! When I executed the code using the Tradingview data rather than my own dataset, the y-axis appeared on the screen at all times - you didnt have to scroll all the way to the right to find it. Do you know if there is a fix for this? it would be nice to be able to see the y-axis when I have not scrolled all the way to the end of the graph. 

My apologies if I wasted your time. Thanks again for being patient with me and for helping (: 

Here's an image of what I mean: 

![y-axis 001](https://github.com/tradingview/lightweight-charts/assets/166934244/46a04c2a-473a-4ffe-be55-9c9cc884fcf2)

**Reply by AdvMaple** [reply]
Here, **I think** you can do 2 thing:
1. call `timeScale.fitContent();` to automatic zoom
2. Set a css `max-height` on the chart container, and ther chart will auto scale

**Reply by TheOxfordDeveloper** [reply]
thanks so much @AdvMaple you've saved the day! The y-axis is now visible without having all of this blank space between the candlesticks and the axis (: 

![visible_yaxis 001](https://github.com/tradingview/lightweight-charts/assets/166934244/cc727d8a-c88a-40eb-ac8e-a84893fb50e8)

**Comment by blackpanther1881** [comment]
I'm also getting issue, my y axix not rendering 

```
"use client";
import { createChart, Time, CandlestickSeries } from "lightweight-charts";
import { useEffect, useRef } from "react";
import { stickData } from "@/Components/data";
import { useApiQuery } from "@/Helpers/useApiQueary";
import * as React from "react";

export const stickData = [
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.0001529350997223",
    high: "0.0001521309074302",
    low: "0.0001521309074302",
    close: "0.0001521309074302",
    volume: "165.3686930996073192",
    startInMilli: 1756533540000,
    endInMilli: 1756533599999,
  },
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.0001528700615685",
    high: "0.0001529350997223",
    low: "0.0001529350997223",
    close: "0.0001529350997223",
    volume: "6.0971311384656274",
    startInMilli: 1756528500000,
    endInMilli: 1756528559999,
  },
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.0001528309084814",
    high: "0.0001528700615685",
    low: "0.0001528700615685",
    close: "0.0001528700615685",
    volume: "6.7863715262944941",
    startInMilli: 1756528380000,
    endInMilli: 1756528439999,
  },
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.0001527760425398",
    high: "0.0001528309084814",
    low: "0.0001528309084814",
    close: "0.0001528309084814",
    volume: "0.9712586759059278",
    startInMilli: 1756528320000,
    endInMilli: 1756528379999,
  },
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.000152716303318",
    high: "0.0001527760425398",
    low: "0.0001527760425398",
    close: "0.0001527760425398",
    volume: "9.9",
    startInMilli: 1756525140000,
    endInMilli: 1756525199999,
  },
  {
    granularity: 60,
    tokenAddress: "0x6d30b4b49a8b0564bf35aeb714417943057804bd",
    open: "0.000152703944063",
    high: "0.000152716303318",
    low: "0.000152716303318",
    close: "0.000152716303318",
    volume: "1.9407322992487796",
    startInMilli: 1756523280000,
    endInMilli: 1756523339999,
  },
  ...
  ]
const CandleStick = ({ agentInfo }: any) => {
  const chartContainerRef: any = useRef();
  const [chartData, setChartData] = React.useState<any>(stickData);

  useEffect(() => {
    if (chartContainerRef?.current?.clientWidth) {
      const chart = createChart(chartContainerRef.current, {
        width: chartContainerRef.current.clientWidth,
        height: 400,
        layout: {
          background: { color: "#171717" },
          textColor: "#ffffff",
        },
        grid: {
          vertLines: { color: "#333" },
          horzLines: { color: "#333" },
        },
        crosshair: {
          mode: 0,
        },
        rightPriceScale: {
          visible: true,
          ticksVisible: true,
          textColor: "#fff",
        },
        timeScale: {
          borderColor: "#333",
          timeVisible: true,
          secondsVisible: false,
        },

        localization: {
          locale: navigator.language,
          timeFormatter: (time: Time) =>
            new Date((time as number) * 1_000).toLocaleString(),
          priceFormatter: (price: number) => {
            if (price < 0.001) {
              return price.toFixed(9); // Show 9 decimal places for very small values
            } else if (price < 0.01) {
              return price.toFixed(6); // Show 6 decimal places for small values
            } else {
              return price.toFixed(4); // Show 4 decimal places for larger values
            }
          },
        },
      });

      const candlestickSeries = chart.addSeries(CandlestickSeries, {
        upColor: "#4CAF50",
        downColor: "#F44336",
        borderDownColor: "#F44336",
        borderUpColor: "#4CAF50",
        wickDownColor: "#F44336",
        wickUpColor: "#4CAF50",
      });

      // Only process data if chartData is available
      if (chartData && chartData.length > 0) {
        const data = chartData
          .map((candle: any) => ({
            time: Math.floor(candle.startInMilli / 1000) as Time,
            open: parseFloat(candle.open),
            high: parseFloat(candle.high),
            low: parseFloat(candle.low),
            close: parseFloat(candle.close),
          }))
          .sort((a: any, b: any) => (a.time as number) - (b.time as number)); // Sort by time in ascending order

        console.log("Chart data:", data);
        candlestickSeries.setData(data);

        if (data.length > 0) {
          const lastTime = data[data.length - 1].time as number;
          const oneDayInSeconds = 24 * 60 * 60;
          const startTime = lastTime - oneDayInSeconds;

          // Set the visible range to show only one day
          chart.timeScale().setVisibleRange({
            from: startTime as Time,
            to: lastTime as Time,
          });

          chart.timeScale().scrollToPosition(0, false);
        }
        chart.timeScale().fitContent();
      }

      const resizeObserver = new ResizeObserver(() => {
        if (chartContainerRef.current) {
          chart.resize(chartContainerRef.current.clientWidth, 400);
        }
      });
      resizeObserver.observe(chartContainerRef.current);

      return () => {
        resizeObserver.disconnect();
        chart.remove();
      };
    }
  }, [chartData]);

  return (
    <div style={{ width: "100%", maxHeight: 400 }}>
      <div
        className="candle-container"
        ref={chartContainerRef}
        style={{
          width: "100%",
          height: 400,
          overflow: "hidden",
          borderRadius: "6px",
          backgroundColor: "#171717",
        }}
      ></div>
    </div>
  );
};
export default CandleStick;
```

this is my component 

<img width="945" height="344" alt="Screenshot 2025-08-30 at 4 37 14 PM" src="https://github.com/user-attachments/assets/5ec49c80-6339-4ebe-854c-801a4bf5d1b9" />
